styles/main.css

```diff
@@ -267,25 +268,25 @@ div.favourites input:checked + label {
 div.gallery {
     display: grid;
     grid-template-columns: 1fr 1fr 1fr 1fr;
+    gap: 10px;
     
 }
 
 div.gallery img {
     display: block;
     width: 100%;
     height: 100%;
-    padding: 5px;
     object-fit: cover;
 }
 div.gallery img:hover {
     scale: 1.1;
 }
 .row-span-2 {
-    grid-row: 1 / span 2;
+    grid-row: span 2;
```

Comment:
Leave the first part, `1 /` off so that the items start wherever they start naturally and span from there.  The 1 before the / causes the element to always start at line 1